pkg/client/kubernetes/clientmap/internal/shoot_clientmap.go

```diff
@@ -123,7 +131,42 @@ func (f *ShootClientSetFactory) NewClientSet(ctx context.Context, k clientmap.Cl
 	return clientSet, err
 }
 
+var _ clientmap.Invalidate = &ShootClientSetFactory{}
+
+func (f *ShootClientSetFactory) InvalidateClient(k clientmap.ClientSetKey) error {
```

Comment:
Can you add a unit test, that ensures that the invalidation logic works (e.g. invalidating a shoot client, changing the cached information and then checking, that the updated information is used)?

Reply:
I extended the already existing tests. I hope that's okay.